tests/unit_tests/ut_midi_stream.cpp

```diff
@@ -0,0 +1,10 @@
+#define CATCH_CONFIG_MAIN
+
+#include <catch2/catch.hpp>
+#include <composer/midi_stream.h>
+#include <MidiFile.h>
+
+TEST_CASE ("MIDI Stream") {
+
```

Comment:
Falta testar suas funções ainda. Para GUI e áudio, realmente é mais difícil testar automaticamente pq tem que escutar. 

Para cada outra função implementada, você pode colocar um unit tests. Por exemplo, use of unit tests para garantir que o is_harmony está funcionando. Explore todas as saídas esperadas. Por exemplo, testar quando dá uma melodia e uma harmonia como entrada. Testar o que acontece com uma entrada vazia. Testar o que acontece com uma entrada que veio de um arquivo.

Para cada função, precisamos dos unit tests que são o que realmente finalizam a coisa como implementada de forma garantida.